packages/neon-core/src/u/HexString.ts

```diff
@@ -0,0 +1,81 @@
+import { ensureHex, reverseHex, hexXor } from "./misc";
+import {
+  str2hexstring,
+  num2hexstring,
+  hexstring2ab,
+  ab2hexstring,
+  hexstring2str
+} from "./convert";
+
+export class HexString {
+  private _value: string;
+  private _littleEndian: boolean;
+
+  protected _checkValue(value: string) {
+    ensureHex(value);
+  }
+
+  protected constructor(value: string, littleEndian = false) {
+    if (value.startsWith("0x")) {
+      value = value.slice(2);
+    }
+    this._checkValue(value);
+    this._value = value;
+    this._littleEndian = littleEndian;
+  }
+
+  /**
+   * Get hex value as string
+   * @param asLittleEndian true, export a little-endian string; false, export a big-endian string; default to be false
+   * @returns string
+   */
+  public value(asLittleEndian = false): string {
```

Comment:
replace `value` and `reverse` with `toLittleEndian` and `toBigEndian`. Alternatively, we can use `toString` and `toReverseString` as the API instead.

Reply:
The `value` can be replaced with `toLittleEndian` and `toBigEndian` but not `reverse`.
`reverse` is a write function to change the value of HexString.

I used `export(asLittleEndian: boolean)` to replace `value` cause I think `toLittleEndian` and `toBigEndian` sounds like they're going to do some conversion to change the value.

Reply:
Remove the write function and we have an immutable class that is very clear in terms of API. If we need to reverse the internal value, it means that we stored the value wrongly. There should never be a need to reverse the value.

We should always be stating how we are using the hexstring values so `toLittleEndian` and `toBigEndian` works nicely.